src/Microsoft.AspNet.OData/Extensions/HttpRequestMessageExtensions.cs

```diff
@@ -416,6 +416,22 @@ public static IEnumerable<IODataRoutingConvention> GetRoutingConventions(this Ht
 
             return request.GetRequestContainer().GetServices<IODataRoutingConvention>();
         }
+
+        /// <summary>
+        /// Gets the <see cref="IExpandQueryBuilder"/> from the request container.
+        /// </summary>
+        /// <param name="request">The request.</param>
+        /// <returns>The <see cref="IExpandQueryBuilder"/> from the request container.</returns>
+        public static IExpandQueryBuilder GetExpandQueryBuilder(this HttpRequestMessage request)
+        {
+            if (request == null)
+            {
+                throw Error.ArgumentNull("request");
+            }
+
+            return request.GetRequestContainer().GetRequiredService<IExpandQueryBuilder>();
```

Comment:
Can we do like: if customer registers one, use it, otherwise, don't do any special things?

Reply:
I prefer we have a default implementation so that the feature works out of the box. If the implementation doesn't work for the customer, they should register their custom implementation